packages/block-editor/src/components/rich-text/index.js

```diff
@@ -453,10 +453,47 @@ ForwardedRichTextContainer.Content = ( {
 	return content;
 };
 
+ForwardedRichTextContainer.Content.__unstableIsRichTextContent = {};
 ForwardedRichTextContainer.isEmpty = ( value ) => {
 	return ! value || value.length === 0;
 };
 
+function findContent( blocks, richTextValues = [] ) {
+	if ( ! Array.isArray( blocks ) ) {
+		blocks = [ blocks ];
+	}
+
+	for ( const block of blocks ) {
+		if (
+			block?.type?.__unstableIsRichTextContent ===
+			ForwardedRichTextContainer.Content.__unstableIsRichTextContent
+		) {
+			richTextValues.push( block.props.value );
+			continue;
+		}
+
+		if ( block?.props?.children ) {
+			findContent( block.props.children, richTextValues );
+		}
+	}
+
+	return richTextValues;
+}
+
+function _getSaveElement( { name, attributes, innerBlocks } ) {
+	return getSaveElement(
+		name,
+		attributes,
+		innerBlocks.map( _getSaveElement )
+	);
+}
+
+export function getRichTextValues( blocks = [] ) {
```

Comment:
What does this function do, can we add some unit test to it?

Reply:
I added some e2e tests, I think that's more valuable